src/segment-loader.js

```diff
@@ -299,7 +317,30 @@ export default class SegmentLoader extends videojs.EventTarget {
       return null;
     }
 
+    // Sanity check the segment-index determining logic above but calcuating
+    // the percentage of the chosen segment that is buffered. If more than 90%
+    // of the segment is buffered then fetching it will likely not help in any
+    // way
+    let percentBuffered = this.getSegmentBufferedPercent_(playlist, mediaIndex, buffered);
+
+    if (percentBuffered >= 90) {
+      // Retry the buffered calculation with the next segment if there is another
+      // segment after the currently selected segment
+      if (mediaIndex + 1 < playlist.segments.length) {
+        percentBuffered = this.getSegmentBufferedPercent_(playlist, mediaIndex + 1, buffered);
+      }
+
+      // If both checks failed return and don't load anything
+      if (percentBuffered >= 90) {
+        return;
```

Comment:
Nah, we should be quiet unless something has actually gone wrong. I think we are too verbose as it is. We'll need to remove some of the logging at some point. (Or hide it behind a debug flag of some kind if it's useful.)